### PR TITLE
[RESTEASY-2257] MP Rest Client SSL support

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
`org.jboss.resteasy.test.response.CompletionStageResponseTest` for OpenJDK 8...

File: resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/RestClientDelegateBean.java [hunk 1]
```diff
@@ -110,10 +135,103 @@ public Object create(CreationalContext<Object> creationalContext) {
...
+
+            builder.trustStore(trustStore);
+        } catch (KeyStoreException e) {
+            throw new IllegalArgumentException("Failed to initialize trust store from " + trustStorePath, e);
+        }
+    }
+
+    private InputStream locateStream(String path) throws FileNotFoundException {
+        if (path.startsWith("classpath:")) {
+            if (path.startsWith("classpath:")) {
```
Comment (asoldano) [inline resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/RestClientDelegateBean.java]: Duplicated check? ;-)

**Comment by asoldano** [comment]
> `org.jboss.resteasy.test.response.CompletionStageResponseTest` failed for OpenJDK 8, server version 16 and no ELYTRON property.
> Worked for everything else...

It's passing after I rescheduled the job.

File: resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/RestClientDelegateBean.java (hunk 1 above)
Comment (michalszynkiewicz) [reply resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/RestClientDelegateBean.java]: fixed, sorry, I missed that when cleaning up

**Comment by michalszynkiewicz** [comment]
@asoldano thank you for the review, should be okay now.

**Pull request merged**